test/uk/gov/hmrc/partnershipidentificationfrontend/service/AuditServiceSpec.scala

```diff
@@ -301,6 +290,74 @@ class AuditServiceSpec extends AnyWordSpec with Matchers with IdiomaticMockito {
       }
     }
+
+    "the user is identifying a limited partnership with all information" when {
+
+      val partnershipFullJourneyData: PartnershipFullJourneyData = PartnershipFullJourneyData(
+        optPostcode = Some(testPostcode),
+        optSautr = Some(testSautr),
+        companyProfile = Some(testCompanyProfile),
+        identifiersMatch = false,
+        businessVerification = Some(BusinessVerificationUnchallenged),
+        registrationStatus = RegistrationNotCalled
+      )
+
+      val journeyConfig = testDefaultGeneralPartnershipJourneyConfig.copy(partnershipType = LimitedPartnership)
+
+      val expectedAuditModel: JsObject = expectedLimitedPartnershipAuditJson(businessType = "Limited Partnership")
+
+      "identifiersMatch is false" should {
+        "audit the correct information" in new Setup {
+          mockPartnershipIdentificationService.retrievePartnershipFullJourneyData(testJourneyId) returns Future.successful(Some(
+            partnershipFullJourneyData
+          ))
+
+          await(TestAuditService.auditPartnershipInformation(
+            journeyId = testJourneyId,
+            journeyConfig = journeyConfig
+          ))
+          mockAuditConnector.sendExplicitAudit("LimitedPartnershipRegistration", expectedAuditModel) was called
+        }
+      }
+      "identifiersMatch is false and service name has not provided" should {
+        "audit the correct information" in new Setup {
+          mockPartnershipIdentificationService.retrievePartnershipFullJourneyData(testJourneyId) returns Future.successful(Some(
+            partnershipFullJourneyData
+          ))
+
+          mockAppConfig.defaultServiceName returns testServiceName
+
+          await(TestAuditService.auditPartnershipInformation(
+            journeyId = testJourneyId,
+            journeyConfig = journeyConfig.copy(pageConfig = testDefaultPageConfig.copy(optServiceName = None))
+          ))
+          mockAuditConnector.sendExplicitAudit("LimitedPartnershipRegistration", expectedAuditModel) was called
+        }
+      }
+    }
   }
-}
 
+  private def expectedAuditJson(partnershipType: String, verificationStatus: String, registerStatus: String, isMatch: Boolean = true): JsObject =
+    Json.obj(
+      "SAUTR" -> testSautr,
+      "SApostcode" -> testPostcode,
+      "isMatch" -> isMatch,
+      "businessType" -> partnershipType,
+      "VerificationStatus" -> verificationStatus,
+      "RegisterApiStatus" -> registerStatus,
+      "callingService" -> testServiceName
+    )
+
+  private def expectedLimitedPartnershipAuditJson(businessType: String): JsObject = {
+    Json.obj(
+      "SAUTR" -> testSautr,
+      "SApostcode" -> testPostcode,
+      "isMatch" -> false,
+      "companyNumber" -> testCompanyProfile.companyNumber,
+      "businessType" -> businessType,
+      "VerificationStatus" -> "Not Enough Information to challenge",
+      "RegisterApiStatus" -> "not called",
+      "callingService" -> testServiceName,
+    )
```

Comment:
I wonder if there's a better way to do this, as we're going to have all of the same scenarios for the incorporated partnership in the future. We could add company profile as an optional field, but that is potentially quite complex for a test method.

Reply:
maybe there are Rob ... we create the second method with Refaque because it is going to use in it task too. .... JsObject have a nice algebra that allow to add and removed field ... I think that is a better approach .... dont know we have to experiment to see how we can fuse the 2 methods in only 1 maybe